File: 1367-linked-list-in-binary-tree/1367-linked-list-in-binary-tree.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool flag;
    bool isit(TreeNode* root,ListNode* head){
        if(head == NULL) return 1;
        if(root == NULL || root->val != head->val) return 0;

        return isit(root->left,head->next) || isit(root->right,head->next);
    }
    void inorder(ListNode* head,TreeNode* root){
        if(root == NULL) return;

        inorder(head,root->left);
        if(root->val == head->val){
            ListNode* temp = head;
            if(isit(root,temp)){
                flag = 1;
                return;
            }
        }
        inorder(head,root->right);
    }
    bool isSubPath(ListNode* head, TreeNode* root) {
        flag = 0;
        inorder(head,root);

        return flag;
    }
};

File: 1367-linked-list-in-binary-tree/1367-linked-list-in-binary-tree.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean flag;
    public boolean isit(TreeNode root,ListNode head){
        if(head == null) return true;
        if(root == null || root.val != head.val) return false;

        return isit(root.left,head.next) || isit(root.right,head.next);
    }
    public void inorder(ListNode head,TreeNode root){
        if(root == null) return;

        inorder(head,root.left);
        // work
        if(root.val == head.val){
            ListNode temp = head;
            if(isit(root,temp)) {
                flag = true;
                return;
            }
        }
        inorder(head,root.right);
    } 
    public boolean isSubPath(ListNode head, TreeNode root) {
        flag = false;
        inorder(head,root);

        return flag;
    }
}